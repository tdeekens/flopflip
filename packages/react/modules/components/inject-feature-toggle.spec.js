import React from 'react';
import { shallow } from 'enzyme';
import injectFeatureToggle from './inject-feature-toggle';

describe('injecting', () => {
  const TestComponent = () =>
    <div>
      {'Test'}
    </div>;
  TestComponent.displayName = 'TestComponent';
  TestComponent.propTypes = {};

  const featureToggle = 'aFeatureToggle';
  let availableFeatureToggles;
  let Component;
  let wrapper;

  beforeEach(() => {
    availableFeatureToggles = { [featureToggle]: true };

    Component = injectFeatureToggle(featureToggle)(TestComponent);
    wrapper = shallow(
      <Component availableFeatureToggles={availableFeatureToggles} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should pass the feature toggle's state as a `prop`", () => {
    expect(wrapper.find(TestComponent)).toHaveProp(
      'featureToggle',
      availableFeatureToggles[featureToggle]
    );
  });
});
