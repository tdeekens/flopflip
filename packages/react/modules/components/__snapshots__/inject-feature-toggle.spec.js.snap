// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`injecting should match snapshot 1`] = `
<TestComponent
  availableFeatureToggles={
    Object {
      "aFeatureToggle": true,
    }
  }
  featureToggle={true}
/>
`;
